File: Solution/257.BinaryTreePaths.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def binaryTreePaths(self, root):
        if not root:
            return []
        result = []
        self.dfs(root, result, "")
        return result

    def dfs(self, node, result, string):
        if not node.left and not node.right:
            result.append(string + str(node.val))
        if node.left:
            self.dfs(node.left, result, string + str(node.val) + "->")
        if node.right:
            self.dfs(node.right, result, string + str(node.val) + "->")
        

File: Solution/Javascript/257. Binary Tree Paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function (root) {
    let res = [];

    let dfs = (node, res, string) => {
        if (node !== null) {
            string += node.val;
        }
        if (!node || (!node.left && !node.right)) {
            if (string.length > 0) {
                res.push(string);
            }
            return;
        }
        string = string + "->";
        if (node.left) {
            dfs(node.left, res, string);
        }
        if (node.right) {
            dfs(node.right, res, string);
        }
    }

    dfs(root, res, "");

    return res;
};